_observability-plugins/data-prepper/data-prepper-reference.md

```diff
@@ -53,7 +52,20 @@ sslKeyFile | Conditionally | String, file-system path or AWS S3 path to the secu
 useAcmCertForSSL | No | Boolean, enables TLS/SSL using certificate and private key from AWS Certificate Manager (ACM). Default is `false`.
 acmCertificateArn | Conditionally | String, represents the ACM certificate ARN. ACM certificate take preference over S3 or local file system certificate. Required if `useAcmCertForSSL` is set to `true`.
 awsRegion | Conditionally | String, represents the AWS region to use ACM or S3. Required if `useAcmCertForSSL` is set to `true` or `sslKeyCertChainFile` and `sslKeyFile` are AWS S3 paths.
+authentication | No | An authentication configuration. By default, this runs an unauthenticated server. This uses pluggable authentication for HTTPS. To provide customer authentication use or create a plugin which implements: [GrpcAuthenticationProvider](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/armeria-common/src/main/java/com/amazon/dataprepper/armeria/authentication/GrpcAuthenticationProvider.java)
+
+### http_source
+
+This is a source plugin that supports HTTP protocol. Currently ONLY support Json UTF-8 codec for incoming request, e.g. `[{"key1": "value1"}, {"key2": "value2"}]`.
 
+Option | Required | Description
+:--- | :--- | :---
+port | No | Integer, the port the source is running on. Default is `2021`. Valid options are between `0` and `65535`.
+request_timeout | No | Integer, the request timeout in millis. Default is `10_000`.
+thread_count | No | Integer, the number of threads to keep in the ScheduledThreadPool. Default is `200`.
+max_connection_count | No | Integer, the maximum allowed number of open connections. Default is `500`.
+max_pending_requests | No | Ingeger, the maximum number of allowed tasks in ScheduledThreadPool work queue. Default is `1024.
+authentication | No | An authentication configuration. By default, this runs an unauthenticated server. This uses pluggable authentication for HTTPS. To provide customer authentication use or create a plugin which implements: [ArmeriaHttpAuthenticationProvider](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/armeria-common/src/main/java/com/amazon/dataprepper/armeria/authentication/ArmeriaHttpAuthenticationProvider.java)
```

Comment:
Do we want to talk about the pluggable details in this reference? This document seems more oriented toward the YAML configuration options. I'd think it would be valuable to document how to use it for `http_basic` in this document.

Reply:
I like the suggestion. Our own documentation is lacking here as well. I will update that in a separate PR.